src/main/java/com/mparticle/kits/ButtonKit.java

```diff
@@ -50,20 +42,45 @@ public ButtonKit getInstance() {
     }
 
     @Override
-    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
+    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, final Context ctx) {
         final String applicationId = settings.get("application_id");
         if (KitUtils.isEmpty(applicationId)) {
-
-            throw new IllegalArgumentException("No Button application ID provided, can't initialize kit.");
+            throw new IllegalArgumentException(
+                    "No Button application ID provided, can't initialize kit."
+            );
         }
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1) {
+            throw new IllegalArgumentException(
+                    "App running in an < API15 environment, can't initialize kit."
+            );
+        }
+
+        ButtonMerchant.configure(getApplicationContext(), applicationId);
+        logDebug("Button Attribution Token: %s", getAttributionToken());
 
-        final HostInformation hostInformation = new HostInformation(context, applicationId);
-        final IdentifierForAdvertiserProvider ifaProvider = new IdentifierForAdvertiserProvider(context);
-        mApi = new ButtonApi(hostInformation, ifaProvider);
-        mStorage = new Storage(context, applicationId);
-        Uri data = MParticle.getInstance().getAppStateManager().getLaunchUri();
-        String action = MParticle.getInstance().getAppStateManager().getLaunchAction();
-        handleIntent(data, action);
+        ButtonMerchant.addAttributionTokenListener(getApplicationContext(),
+                new ButtonMerchant.AttributionTokenListener() {
+                    @Override
+                    public void onAttributionTokenChanged(@NonNull String s) {
+                        logDebug("Refreshed Button Attribution Token: %s", s);
+                    }
+                });
+
+        ButtonMerchant.handlePostInstallIntent(getApplicationContext(),
+                new PostInstallIntentListener() {
+                    @Override
+                    public void onResult(@Nullable Intent intent, @Nullable Throwable throwable) {
+                        PackageManager pm = getApplicationContext().getPackageManager();
+                        if (intent != null && intent.resolveActivity(pm) != null) {
+                            logDebug("Handling post-install intent for %s", intent.toString());
+                            getApplicationContext().startActivity(intent);
+                        } else if (throwable != null) {
+                            logError("Error checking post install intent", throwable);
+                        } else {
+                            logError("Neither an Intent nor a Throwable is available post-install");
```

Comment:
This is not an error state but expected if the request successfully completes and no post install deep link was found.